### PR TITLE
[c++] Fix some container issues

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

As title sais. No real functionality was changed.

## Steps to test these changes

None.

### CODE REVIEW

**Review by TeoTwawki — APPROVED** [review]

**Review by claywar — APPROVED** [review]

File: src/map/packet_system.cpp [hunk 1]
```diff
@@ -2540,7 +2540,18 @@ void SmallPacket0x04D(map_session_data_t* const PSession, CCharEntity* const PCh
...
 
-            if (quantity > 0 && PItem && PItem->getQuantity() >= quantity && PChar->UContainer->IsSlotEmpty(slotID))
+            if (quantity == 0 || !PItem)
+            {
+                return;
+            }
+
+            if (PItem->getQuantity() < quantity || PItem->getReserve() > 0)
+            {
+                ShowWarning("Delivery Box: %s attempted to send insufficient/reserved %u %s (%u).", PChar->getName(), quantity, PItem->getName(), PItem->getID());
```
Comment (Frankie-hz) [inline src/map/packet_system.cpp:2550]: I believe `PChar->getName()` should be `PChar->GetName()`

File: src/map/entities/charentity.cpp [hunk 2]
```diff
@@ -2079,6 +2079,14 @@ void CCharEntity::OnItemFinish(CItemState& state, action_t& action)
     auto* PTarget = static_cast<CBattleEntity*>(state.GetTarget());
     auto* PItem   = state.GetItem();
 
+    if (!PItem->isType(ITEM_EQUIPMENT) && (PItem->getQuantity() < 1 || PItem->getReserve() > 0))
+    {
+        ShowWarning("OnItemFinish: %s attempted to use reserved/insufficient %s (%u).", this->getName(), PItem->getName(), PItem->getID());
```
Comment (Frankie-hz) [inline src/map/entities/charentity.cpp:2084]: I believe `this->getName()` should be `this->GetName()`

getName() looks like an item string

Comment (zach2good) [reply src/map/entities/charentity.cpp:2084]: We renamed entity and zone GetName's to getName, and we plan on doing the same for everything else so all member functions (and their bindings) start with lower case letters 👍

Comment (Frankie-hz) [reply src/map/entities/charentity.cpp:2084]: ahh ok, thank you!

**Pull request merged**